packages/palette/src/elements/Shelf/Shelf.tsx

```diff
@@ -0,0 +1,289 @@
+import { themeGet } from "@styled-system/theme-get"
+import React, {
+  Children,
+  createRef,
+  isValidElement,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react"
+import styled from "styled-components"
+import { useCursor } from "use-cursor"
+import { visuallyDisableScrollbar } from "../../helpers/visuallyDisableScrollbar"
+import { ChevronIcon } from "../../svgs"
+import { Box, BoxProps } from "../Box"
+import { CELL_GAP_PADDING_AMOUNT, paginateCarousel } from "../Carousel"
+import { CarouselBar } from "../CarouselBar"
+import { Clickable } from "../Clickable"
+import { FullBleed } from "../FullBleed"
+
+/** ShelfProps */
+export type ShelfProps = BoxProps & {
+  snap?: "none" | "start" | "end" | "center"
+  children: JSX.Element | JSX.Element[]
+  onChange?(index: number): void
+  onProgress?(progress: number): void
+}
+
+/**
+ * A Shelf is a new kind of carousel...
+ */
+export const Shelf: React.FC<ShelfProps> = ({
+  children,
+  snap = "none",
+  onChange,
+  onProgress,
+  ...rest
+}) => {
+  const cells = useMemo(
+    () =>
+      Children.toArray(children)
+        .filter(isValidElement)
+        .map((child) => ({ child, ref: createRef<HTMLLIElement>() })),
+    [children]
+  )
+
+  const containerRef = useRef<HTMLDivElement | null>(null)
+  const viewportRef = useRef<HTMLDivElement | null>(null)
+
+  const [mounted, setMounted] = useState(false)
+  const [pages, setPages] = useState([0])
+  const [progress, setProgress] = useState(0)
+  const [offset, setOffset] = useState(0)
+
+  const init = () => {
+    if (containerRef.current === null || viewportRef.current === null) return
+
+    // Set page-stops
+    const { current: viewport } = viewportRef
+    const values = cells.map(({ ref }) => ref.current.clientWidth)
+    setPages(paginateCarousel({ viewport: viewport.clientWidth, values }))
```

Comment:
This is one of those times the fix I was just looking for turned out to be simple: here I was passing in the viewport width to calculate the page-stops. Now that the viewport is full-bleed we actually want the container width which specifies that more narrow window. It works 👀  !

Reply:
Lol, I think I spoke too soon. I'm close though.